### PR TITLE
Brain imaging improvements

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Improved image rendering and handled multiple images. Added sliders on top. On download, download images resized. Charts table to select the sample taking less space by removing column width. Also replaced 'python3' with serverconfig.python. Read image size from dataset to avoid delays on client. See corresponding PR in [sjpp](https://github.com/stjude/sjpp/pull/539)

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by congyu-lu — APPROVED** [review]

File: server/routes/brainImaging.ts [hunk 1]
```diff
@@ -51,60 +74,100 @@ async function getBrainImage(query: GetBrainImagingRequest, genomes: any) {
...
+		const cmd = [
+			`${serverconfig.binpath}/../python/src/plotBrainImaging.py`,
+			refFile,
+			plane,
+			index,
+			showLegend,
+			maxLength,
+			filesJson
+		]
+		console.log(cmd.join(' '))
```
Comment (xzhou82) [inline server/routes/brainImaging.ts:151]: please delete log

File: server/routes/brainImaging.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import fs from 'fs'
 import path from 'path'
 import serverconfig from '#src/serverconfig.js'
-import type { GetBrainImagingRequest, GetBrainImagingResponse } from '#types'
+import type { CategoricalTW, GetBrainImagingRequest, FilesByCategory, TermWrapper } from '#types'
 import { spawn } from 'child_process'
+import { getData } from '../src/termdb.matrix.js'
+import { get } from 'http'
```
Comment (xzhou82) [inline server/routes/brainImaging.ts:7]: unused import

File: server/routes/brainImaging.ts [hunk 3]
```diff
@@ -18,6 +20,15 @@ export const api: any = {
 			response: {
 				typeId: 'GetBrainImagingResponse'
 			}
+		},
+		post: {
```
Comment (xzhou82) [inline server/routes/brainImaging.ts:24]: combine replace get{} and post{} with all{}

File: server/src/mds3.init.js [hunk 4]
```diff
@@ -1814,7 +1814,7 @@ async function validate_query_NIdata(ds, genome) {
 				}
 
 				return new Promise((resolve, reject) => {
-					const ps = spawn('python3', [
+					const ps = spawn(serverconfig.python, [
```
Comment (xzhou82) [inline server/src/mds3.init.js:1817]: should move validate_query_NIdata() from mds3.init.js into server/routes/brainImagingSamples.ts, so to keep the logic together

File: shared/types/src/routes/brainImaging.ts [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
-	/** when true will only return all the samples that have NIdata */
-	samplesOnly?: boolean
 	/** the slice index of sagittal, coronal and axial planes*/
 	l?: string
 	f?: string
 	t?: string
 	/** the sample names selected by the users to plot on brain template */
-	selectedSampleFileNames?: string[]
+	selectedSampleFileNames: string[]
+	divideByTW?: CategoricalTW
```
Comment (xzhou82) [inline shared/types/src/routes/brainImaging.ts:16]: comment

File: shared/types/src/routes/brainImaging.ts [hunk 6]
```diff
@@ -1,21 +1,41 @@
...
 }
 
+export type GetBrainImagingSamplesRequest = {
+	/** a user-defined genome label in the serverconfig.json, hg38, hg19, mm10, etc */
+	genome: string
+	/** a user-defined dataset label in the serverconfig.json, such as ClinVar, SJLife, GDC, etc */
+	dslabel: string
+	/** a user-defined brain template label in dataset file, such as Ref1, Ref2 */
+	refKey: string
+	/** when true will only return all the samples that have NIdata */
```
Comment (xzhou82) [inline shared/types/src/routes/brainImaging.ts:27]: delete old comment

File: server/routes/brainImaging.ts (hunk 3 above)
Comment (siosonel) [reply server/routes/brainImaging.ts:24]: @xzhou82 I told Irene to declare `post` separated. As part of refactoring the routes code, I found out that it's not good to use `all`, it results in 20+ route methods being set besides `get` and `post`. For example, if I console.log the route objects from the `express` app, I see `head`, `patch`, `delete`, and many other methods being set, when we only want to set `get` and `post`.

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**